cishouseholds/pipeline/survey_responses_version_2_ETL.py

```diff
@@ -109,42 +109,60 @@ def transform_survey_responses_generic(df: DataFrame) -> DataFrame:
     df = assign_ethnicity_white(
         df, column_name_to_assign="ethnicity_white", ethnicity_group_column_name="ethnicity_group"
     )
-    df = assign_column_to_date_string(
-        df=df,
-        column_name_to_assign="visit_date_string",
-        reference_column="visit_datetime",
-        time_format="ddMMMyyyy",
-        lower_case=True,
-    )
-    df = assign_column_to_date_string(
-        df=df,
-        column_name_to_assign="visit_datetime",
-        reference_column="visit_datetime",
-        time_format="ddMMMyyyy HH:mm:ss",
-        lower_case=True,
-    )
-    df = assign_column_to_date_string(
-        df=df,
-        column_name_to_assign="samples_taken_date_string",
-        reference_column="samples_taken_datetime",
-        time_format="ddMMMyyyy",
-        lower_case=True,
-    )
-    df = assign_column_to_date_string(
-        df=df,
-        column_name_to_assign="samples_taken_datetime_string",
-        reference_column="samples_taken_datetime",
-        time_format="ddMMMyyyy HH:mm:ss",
-        lower_case=True,
-    )
+    # dict: key=column_name, value=origin
+    date_format_dict = {
+        "visit_date_string": "visit_datetime",
+        "samples_taken_date_string": "samples_taken_datetime",
+    }
+    datetime_format_dict = {
+        "visit_datetime_string": "visit_datetime",
+        "samples_taken_datetime_string": "samples_taken_datetime",
+        # "improved_visit_datetime_string": "improved_visit_date", # TODO: uncomment when improved_visit_date is generated
+    }
+    date_format_string_list = [
+        "date_of_birth",
+        # "improved_visit_date", # TODO: uncomment when generated
```

Comment:
The commented out variables only exist on some of the input files. Moving these dictionaries and loops to the union dependent transformation function should fix the issue. However, do run this in DevTest to ensure that it runs, as it isn't covered by integration tests